src/main/java/com/uber/cadence/context/OpenTelemetryContextPropagator.java

```diff
@@ -0,0 +1,144 @@
+      W3CTraceContextPropagator.getInstance();
+  private static final TextMapPropagator w3cBaggagePropagator = W3CBaggagePropagator.getInstance();
+  private static ThreadLocal<Scope> currentContextOtelScope = new ThreadLocal<>();
+  private static ThreadLocal<Span> currentOtelSpan = new ThreadLocal<>();
+  private static ThreadLocal<Scope> currentOtelScope = new ThreadLocal<>();
+  private static ThreadLocal<Iterable<String>> otelKeySet = new ThreadLocal<>();
+  private static final TextMapSetter<Map<String, String>> setter = Map::put;
+  private static final TextMapGetter<Map<String, String>> getter =
+      new TextMapGetter<Map<String, String>>() {
+        @Override
+        public Iterable<String> keys(Map<String, String> carrier) {
+          return otelKeySet.get();
+        }
+
+        @Nullable
+        @Override
+        public String get(Map<String, String> carrier, String key) {
+          return MDC.get(key);
+        }
+      };
+
+  @Override
+  public String getName() {
+    return "OpenTelemetry";
+  }
+
+  @Override
+  public Map<String, byte[]> serializeContext(Object context) {
+    Map<String, byte[]> serializedContext = new HashMap<>();
+    Map<String, String> contextMap = (Map<String, String>) context;
+    if (contextMap != null) {
+      for (Map.Entry<String, String> entry : contextMap.entrySet()) {
+        serializedContext.put(entry.getKey(), entry.getValue().getBytes(Charset.defaultCharset()));
+      }
+    }
+    return serializedContext;
+  }
+
+  @Override
+  public Object deserializeContext(Map<String, byte[]> context) {
+    Map<String, String> contextMap = new HashMap<>();
+    for (Map.Entry<String, byte[]> entry : context.entrySet()) {
+      contextMap.put(entry.getKey(), new String(entry.getValue(), Charset.defaultCharset()));
+    }
+    return contextMap;
+  }
+
+  @Override
+  public Object getCurrentContext() {
+    Map<String, String> carrier = new HashMap<>();
+    w3cTraceContextPropagator.inject(Context.current(), carrier, setter);
+    w3cBaggagePropagator.inject(Context.current(), carrier, setter);
+    return carrier;
+  }
+
+  @Override
+  public void setCurrentContext(Object context) {
+    Map<String, String> contextMap = (Map<String, String>) context;
+    if (contextMap != null) {
+      for (Map.Entry<String, String> entry : contextMap.entrySet()) {
+        MDC.put(entry.getKey(), entry.getValue());
+      }
+      otelKeySet.set(contextMap.keySet());
+    }
+  }
+
+  @Override
+  @SuppressWarnings("MustBeClosedChecker")
+  public void setUp() {
+    Context context =
+        Baggage.fromContext(w3cBaggagePropagator.extract(Context.current(), null, getter))
+            .toBuilder()
+            .build()
+            .storeInContext(w3cTraceContextPropagator.extract(Context.current(), null, getter));
+
+    currentContextOtelScope.set(context.makeCurrent());
+
+    Span span =
+        GlobalOpenTelemetry.getTracer("cadence-client")
+            .spanBuilder("cadence.workflow")
```

Comment:
Not familar with opentelemtry -- How are these two properties used? I wonder if we should avoid harhcoding here because users may need to customize them.